kedro/framework/cli/starters.py

```diff
@@ -116,14 +116,27 @@ class KedroStarterSpec:  # noqa: too-few-public-methods
 kedro new --addons=none
 """
 
-ADD_ONS_DICT = {
+ADD_ONS_SHORTNAME_TO_NUMBER = {
+    "lint": "1",
+    "test": "2",
+    "log": "3",
+    "docs": "4",
+    "data": "5",
+    "pyspark": "6",
+}
+NUMBER_TO_ADD_ONS_NAME = {
     "1": "Linting",
     "2": "Testing",
     "3": "Custom Logging",
     "4": "Documentation",
     "5": "Data Structure",
     "6": "Pyspark",
 }
+ADD_ONS_SHORTNAME_TO_NAME = {
+    short_name: NUMBER_TO_ADD_ONS_NAME[num]
+    for short_name, num in ADD_ONS_SHORTNAME_TO_NUMBER.items()
+}
+
```

Comment:
Nit: This seems unnecessary? You can just use `list(ADD_ONS_SHORTNAME_TO_NUMBER)` below.
```suggestion
```

Reply:
Yeah actually initially I was thinking I can skip the number and just convert from the short name to the longer one. It's a bit tricky and since there are other PR working on this I don't want to create too much merge conflict.

I took your suggestion and remove this extra dict for now, will revisit.